src/components/molecules/OakCopyLinkButton/OakCopyLinkButton.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from "react";
+
+import { OakSmallSecondaryButton } from "@/components/molecules";
+import { OakBox } from "@/components/atoms";
+
+type OakCopyLinkButtonProps = {
+  /**
+   * Href of the link that should be copied
+   */
+  href?: string;
+};
+
+/**
+ * Display copy link button
+ */
+export const OakCopyLinkButton = ({ href }: OakCopyLinkButtonProps) => {
+  const [label, setLabel] = useState("Copy link");
+  const [active, setActive] = useState(true);
+  const [announce, setAnnounce] = useState("");
+
+  const copyLink = () => {
+    if (navigator.clipboard) {
+      const urlToCopy = href || window.location.href;
+      navigator.clipboard.writeText(urlToCopy);
+
+      const copyMessage = "Link copied";
+      setLabel(copyMessage);
+      setAnnounce(copyMessage);
+      setActive(false);
+
+      setTimeout(() => {
+        setAnnounce(""); // used for aria-live announcement
+        setActive(true);
+        setLabel("Copy link");
+      }, 4000);
```

Comment:
why this amount of time?

Reply:
a request from @Monika-Koziol and @RBrownsett in the comment above, let me put it in the const so it is more clear what is this timer for